foursight_core/environment.py

```diff
@@ -63,12 +63,16 @@ def list_unique_environment_names(self) -> List[EnvName]:
             result.add(infer_foursight_from_env(envname=env))
         return sorted(result)  # a list and sorted
 
+    cache_list_environment_names = None
```

Comment:
This initialization should be done in the `__init__` method. What you did would work but is not the normal way to do this.

Reply:
I also would call this `cached...` not `cache_...` because the first word is an adjective.

Reply:
Probably I would upcase the cache, even though it's not constant, if you wanted to do it as a class var.
Then do:
```
@classmethod set_environment_names_cache(cls, value):
    cls.CACHED_ENVIRONMENT_NAME_LIST = value
```
and use that method to set the cache.  Then in the instance method you'd check that variable or call `self.set_environment_names_cache` to set it causing it to get set at class level. If you wanted to set it directly you have to do self.__class__.CACHED_ENVIRONMENT_NAME_LIST = value to have it set at the class level not the instance level. Doing it at class level would keep multiple instances of this class from having to get new caches.